src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs

```diff
@@ -132,6 +132,12 @@ private Task OnStateChangedAsync(INetworkPeer peer, NetworkPeerState previousSta
                     this.peerAddressManager.PeerHandshaked(peer.PeerEndPoint, this.dateTimeProvider.GetUtcNow());
             }
 
+            if ((peer.Inbound) && (peer.State == NetworkPeerState.HandShaked) &&
+                (this.Mode == PeerAddressManagerBehaviourMode.Advertise || this.Mode == PeerAddressManagerBehaviourMode.AdvertiseDiscover))
+            {
+                this.peerAddressManager.AddPeer(peer.PeerEndPoint, IPAddress.Loopback);
```

Comment:
Not sure about the Loopback @dangershony ?

Reply:
The problem here is with `peer.PeerEndPoint` this might not be correct this is the endpoint the socket is connected on so the port can be incorrect.

Reply:
OK so PeerEndPoint here is set to:

```
client.Client.RemoteEndPoint
```

in

`public INetworkPeer CreateNetworkPeer(TcpClient client, NetworkPeerConnectionParameters parameters = null, NetworkPeerDisposer networkPeerDisposer = null)`

This is not correct?

Reply:
@dangershony Im going to add some extra logging in key places so that we can see what the respective values are. Will post my findings here.